CONTRIBUTING.md

```diff
@@ -28,3 +36,69 @@ To update the go dependencies for bazel using `go.mod`:
 ```shell
 gazelle-update-repos
 ```
+
+## pre-commit
+
+It is strongly recommended to run tests locally before you commit or a push a
+change. The simplest way to do that is to use
+[pre-commit](https://pre-commit.com/). This will automatically run all linters
+and tests as a git pre-commit hook.
+
+```shell
+$ pre-commit install --install-hook
+pre-commit installed at .git/hooks/pre-commit
+[INFO] Initializing environment for https://github.com/executablebooks/mdformat.
+[INFO] Initializing environment for https://github.com/executablebooks/mdformat:mdformat-black,mdformat-gfm.
+[INFO] Initializing environment for https://github.com/pre-commit/pre-commit-hooks.
+[INFO] Installing environment for https://github.com/executablebooks/mdformat.
+[INFO] Once installed this environment will be reused.
+[INFO] This may take a few minutes...
+[INFO] Installing environment for https://github.com/pre-commit/pre-commit-hooks.
+[INFO] Once installed this environment will be reused.
+[INFO] This may take a few minutes...
+```
+
+## Visual Studio Code
+
+In order for [VS Code](https://code.visualstudio.com) you will need to configure
+it to use a bazel-aware _gopackagesdriver_ with the gopls language server. To do
+that use the following for `.vscode/settings.json`, replacing
+`/Users/wichert/Code/go-zserio` with the correct location on your machine.
+
+```json
+{
+  "go.goroot": "/Users/wichert/Code/go-zserio/bazel-go-zserio/external/go_sdk",
+  "go.toolsEnvVars": {
+    "GOPACKAGESDRIVER": "/Users/wichert/Code/go-zserio/bin/gopackagesdriver.sh"
```

Comment:
Why don't we use the `{workspaceFolder}` variable in VSCode as mentioned in https://github.com/bazelbuild/rules_go/wiki/Editor-setup#visual-studio-code? That way people can just copy paste the config and not worry. :)

Reply:
Because it doesn't actually work: microsoft/vscode#2809